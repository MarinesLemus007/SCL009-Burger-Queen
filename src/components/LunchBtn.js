import React from 'react';
import Btn from './Btn'
import Menu from '../data/menu.json'
import '../index.css';

class LunchBtn extends React.Component{
    
    render(){
        return (
            <>
            
        <div id="lunch" className="item-btn-row">
            <div className="content-item-btn-row">
                <h4>Hamburguesas</h4>
                <div className="item-btn-row">
                    {Menu.Almuerzos.Hamburguesas.Simple.map(btn=><Btn name={btn.name} value={btn.value} add={this.props.add} key={btn.name}/>)}
                </div>
                <div className="item-btn-row">
                    {Menu.Almuerzos.Hamburguesas.Doble.map(btn=><Btn name={btn.name} value={btn.value} add={this.props.add} key={btn.name}/>)}
                </div>
                <div className="item-btn-row">
                    {Menu.Almuerzos.Agregados.map(btn=>
                    <button className="btn-item" onClick={() => this.props.addCheeseorEgg(btn.name)}key={btn.name}>{btn.name.toUpperCase()} </button>)}
                </div>
            </div>
            <div className="content-item-btn-row">
                <h4>Acompañamientos</h4>
                <div className="item-btn-row">
                    {Menu.Almuerzos.Acompañamientos.map(btn=><Btn name={btn.name} value={btn.value} add={this.props.add} key={btn.name}/>)}
                </div>
            </div>
            <div className="content-item-btn-row">
                <h4>Bebestibles</h4>
                <div className="item-btn-row">
                    {Menu.Almuerzos.Bebestibles.Agua.map(btn=><Btn name={btn.name} value={btn.value} add={this.props.add} key={btn.name}/>)}
                </div>
                <div className="item-btn-row">
                    {Menu.Almuerzos.Bebestibles.Gaseosa.map(btn=><Btn name={btn.name} value={btn.value} add={this.props.add} key={btn.name}/>)}
                </div>
            </div>
        </div>
    </>)
    }
}

export default LunchBtn;