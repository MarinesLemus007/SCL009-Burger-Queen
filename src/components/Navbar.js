import React from 'react';
import logo from '../img/logo.png';
import { BrowserRouter as Router, Route, Link } from "react-router-dom"

class Navbar extends React.Component{
constructor(props){
super(props)
this.state = {};
}

render(){
    return (
      

<header>
    <nav className="menu-nav-row">
        <div className="menu-left-col">
            <img className="menu-icon" src={logo} alt="logo" />
        </div>
        <div className="menu-center-col">
            <ul className="menu-ul-row">
                <li className={this.props.state === 'menu' ? 'btn-selected-navbar' : 'menu-ul'}><Link to="/">MENÃš</Link></li>
                <li className={this.props.state === 'kitchen' ? 'btn-selected-navbar' : 'menu-ul'}><Link to="/Kitchen">COCINA</Link></li>
                <li className={this.props.state === 'ordersReady' ? 'btn-selected-navbar' : 'menu-ul'}><Link to="/OrdersReady">PEDIDOS LISTOS</Link></li>
                <li className="menu-ul"><Link to="/">HISTORIAL</Link></li>
            </ul>
        </div>
        <div className="menu-right-col">
        </div>
    </nav>
</header>

)

}}
export default Navbar;