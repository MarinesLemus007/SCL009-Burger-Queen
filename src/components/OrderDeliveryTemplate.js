import React from 'react';
class OrderDeliveryTemplate extends React.Component{
   render(){
    return (
         this.props.data !== undefined ? this.props.data.map(el=>
            
            
            <div className='order-kitchen' key={el.data.id}>
                <p>Cliente : {el.data.client}</p>
                <p>Fecha :  {new Date(el.data.time).toLocaleDateString()} {new Date(el.data.time).toLocaleTimeString([], {hour: '2-digit', minute:'2-digit'})}</p>
                <p>Tiempo que tomo el pedido : {new Date(el.data.time_ready).toLocaleTimeString([], {hour: '2-digit', minute:'2-digit'}) }</p>
                {el.data.list.map(e=> 
                 <p key={e.name+e.id}>{e.count} x {e.name}</p> )}
                <div className='template-order'>
                    <button className='btn-order-template' onClick={()=>this.props.changeReadyStatus(el.data.id)}>Entregado</button>
                </div>
            </div> 
        
            ):<p></p> 

       )
   }
}
export default OrderDeliveryTemplate;