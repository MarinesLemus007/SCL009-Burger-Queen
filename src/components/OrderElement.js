import React from 'react';
class OrderElement extends React.Component{
   render(){
     return (
       <div className='order-element-row'>
       <div className='order-element-count-col'>
           <p>{this.props.count}</p>
       </div>
       <div className='order-element-name-col'>
         <p>{this.props.name} </p>
       </div>
       <div className='order-element-value-col'>
       <p>${this.props.value}</p>
       </div>
       <div className='order-element-btn-col'>
       <button className="btn-delete" onClick={() => this.props.delete(this.props.id)}>X</button>
       </div>
       </div>
     )
   }
}
export default OrderElement;