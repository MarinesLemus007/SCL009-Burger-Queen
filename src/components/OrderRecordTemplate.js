import React from 'react';
class OrderRecordTemplate extends React.Component{
 
   render(){
    return (
         this.props.data !== undefined ? this.props.data.map(el=>
            
            <div className='order-record' key={el.data.id}>
               
         
                <p>{"Fecha :  "+new Date(el.data.time).toLocaleDateString()}</p>
                <p>{"Hora de pedido : "+new Date(el.data.time).toLocaleTimeString([], {hour: '2-digit', minute:'2-digit'})}</p>
                <p>{"Tiempo en cocina : "+new Date(el.data.time_ready - el.data.time).getMinutes()+" min" }</p>
                <p>{"Entregado a las : "+new Date(el.data.time_delivered).toLocaleTimeString([], {hour: '2-digit', minute:'2-digit'})}</p>
                <p>{"Cliente : "+el.data.client}</p>
                {el.data.list.map(e=> 
                 <p key={e.name+e.id}>{e.count+"  x  "+e.name}</p> )}
                
            </div> 
            ):<p></p> 

       )
   }
}
export default OrderRecordTemplate;