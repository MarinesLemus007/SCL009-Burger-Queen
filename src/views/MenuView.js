import React from 'react';
import {db} from '../data/firebase';
import '../index.css';
import Navbar from '../components/Navbar';
import Btn from '../components/Btn'
import Menu from '../data/menu.json'
import Order from '../components/Order';
import OrderName from '../components/OrderName';
import CategoryBtn from '../components/CategoryBtn'
import LunchBtn from '../components/LunchBtn'


class MenuView extends React.Component{
  
  constructor(props){
    super(props)
   
    this.state = {list: [], client:"",selectedNavbar:'menu',errorMsg:''};
    this.add = this.add.bind(this);
    this.delete = this.delete.bind(this);
    this.view = this.view.bind(this);
    this.changeClient = this.changeClient.bind(this)
    this.clearOrder = this.clearOrder.bind(this);
    this.saveOrder = this.saveOrder.bind(this);
    this.addCheeseorEgg = this.addCheeseorEgg.bind(this);
    this.index = 0; // id de cada elemento de orden creado
   
  }
  //aÃ±ade un producto({name,value,id} a la orden(list -> array de objetos [el index comienza en 0 y se utiliza como id]
  add(valueToAdd,nameToAdd){
    let newList = [...this.state.list] 
          for( let i = 0; i < newList.length; i++){ 
            if ( newList[i].name === nameToAdd) {
                newList[i].count++;
                newList[i].value = newList[i].value + valueToAdd;
              return this.setState({list:newList})
            }
          }
    this.setState({list: this.state.list.concat([{name : nameToAdd, unit_value:valueToAdd, value:valueToAdd, id:this.index, count:1 }])});
    this.index ++;
  }

  addCheeseorEgg(eggOrCheese){
    let newList = [...this.state.list] 
    let lastItem = newList[newList.length-1];
    if(lastItem ===undefined){
      return
    }
    if ( eggOrCheese === '+ Queso') {
    
              lastItem.name = lastItem.name+" + Queso"
              lastItem.value = lastItem.value + 500;
              return this.setState({list:newList})
    }
              lastItem.name = lastItem.name+" + Huevo"
              lastItem.value = lastItem.value + 500;
             this.setState({list:newList})
  }
  

  //
  delete(id){
    let newList = [...this.state.list] //crea copia de list para no cambiar directamente el estado de un componente
    for( var i = 0; i < newList.length; i++){ //recorre la newList y consigue la id de cada elemento,si coincide
      if ( newList[i].id === id) { 
        if( newList[i].count !== 1){
          newList[i].value = newList[i].value - newList[i].unit_value ;
          newList[i].count--;
          return this.setState({list:newList})
        }
        newList.splice(i, 1); //con la misma id que se le pasa borra este elemento (desde posicion i, elimina 1 elemento)
        break;
      }
    }
   this.setState({list:newList}) //remplaza la list con la newList modificada
  }


  view(category){ // dependiendo de la categoria cambia el estado, este metodo se pasa como propiedad al boton
    category = category.toLowerCase()
      
    if(category ==="desayunos"){
      this.setState({
        desayunos:true,
        almuerzos:false
        })
    }
    if(category ==="almuerzos"){
      this.setState({
        desayunos:false,
        almuerzos:true
        })
      }
    }

  changeClient(el){
    this.setState({
      client: el.target.value
    })
  }

  clearOrder(){
    this.setState({
      list: [],
      client:"",
      errorMsg:""
    })
  }

  saveOrder(){
    if(this.state.client === ""){
      this.setState({
        errorMsg:'Debe ingresar un nombre.'}) 
      return 
    }
    let idClient =this.state.client + Date.now();

    let data=
    {
    id: idClient,
    client: this.state.client,
    list: this.state.list,
    not_ready: true,
    time: Date.now() 
    }

    db.collection("ordenes").doc(idClient).set(data)
    .then(() => {
        this.clearOrder();
    })

  }



  render(){
    return (
      <>
        <Navbar state={this.state.selectedNavbar} />
          <div className="content-row">
            <section className="button-content-col">
              <div className="category-btn-row">
                {Object.keys(Menu).map(btn=> <CategoryBtn name={btn.toUpperCase()} state={btn === 'Desayunos' ? this.state.desayunos : this.state.almuerzos} view={this.view} key={btn}/>)}
              </div>
                {this.state.desayunos &&
                <div className="item-btn-row"> {Menu.Desayunos.map(btn=><Btn name={btn.name} value={btn.value} add={this.add} key={btn.name}/>)}
                </div> }
                {this.state.almuerzos && <LunchBtn add={this.add} addCheeseorEgg={this.addCheeseorEgg}/>}  
            </section>  
            <aside className="side-content-col">
            <div className="aside-content">
              <div className="line-order">
                <p>PEDIDO</p>
                <p className='error-msg'>{this.state.errorMsg}</p>
              </div>
              
              <OrderName changeClient={this.changeClient} client={this.state.client}/>
              <div className="order-content">
                <Order list = {this.state.list} delete={this.delete}/>
              </div>
              <footer className="footer-side">
                <div className="btn-clear-col">
                  <button className="btn-aside-clear" onClick={this.clearOrder}>LIMPIAR</button>
                </div>
                <div className="btn-send-col">
                  <button className="btn-aside" onClick={this.saveOrder}>ENVIAR </button>
                </div>
              </footer>
              </div>
            </aside>
          </div>
      </>
    )
  }
}

export default MenuView;

