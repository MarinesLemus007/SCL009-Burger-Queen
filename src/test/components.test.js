import React from 'react';
import { configure , shallow , mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MenuView from '../views/MenuView'
import LunchBtn from '../components/LunchBtn'
<<<<<<< HEAD
import Btn from '../components/Btn'
=======
>>>>>>> b1f7913b807eed429ee47223ea9cf615e1eee423
import CategoryBtn from '../components/CategoryBtn';
import OrderTemplate from '../components/OrderTemplate'
import OrderName from '../components/OrderName';
import OrderElement from '../components/OrderElement';
import OrderRecordTemplate from '../components/OrderRecordTemplate';
import toJson from 'enzyme-to-json';
import OrderDeliveryTemplate from '../components/OrderDeliveryTemplate';
import Navbar from '../components/Navbar';
import Btn from '../components/Btn';

configure({ adapter: new Adapter() });

//tests que verifica si existen los componentes
describe('<MenuView/>', () => {
  test('deberia retornar que el componente devuelve un valor ', () => {
      const wrapper = shallow(<MenuView/>);
      expect(wrapper.find(MenuView)).toBeDefined();//verifica que el componente este devolviendo "algo"
  });
});
describe('<CategoryBtn/>', () => {
  test('deberia retornar que el componente existe y devuelve un valor', () => {
      const wrapper = shallow(<CategoryBtn/>);
      expect(wrapper.find(CategoryBtn)).toBeDefined();
  });
});
describe('<LunchBtn/>', () => {
  test('deberia retornar que el componente existe devuelve un valor', () => {
      const wrapper = shallow(<LunchBtn/>);
      expect(wrapper.find(LunchBtn)).toBeDefined();
  });
});
describe('<OrderElement/>', () => {
    test('deberia retornar que el componente existe devuelve un valor', () => {
        const wrapper = mount(<OrderElement/>);
        expect(wrapper.find(OrderElement)).toBeDefined();
    });
  });
  describe('<OrderTemplate/>', () => {
    test('deberia retornar que el componente existe y devuelve un valor', () => {
        const wrapper = mount(<OrderTemplate/>);
        expect(wrapper.find(OrderElement)).toBeDefined();
    });
  });
  describe('<OrderName/>', () => {
    test('deberia retornar que el componente existe y devuelve un valor', () => {
        const wrapper = mount(<OrderName/>);
        expect(wrapper.exists()).toBe(true);
    });
  });
 


it('deberia ejecutar saveOrder para el boton enviar con la clase dada ', () => {  
    const wrapper = shallow(<MenuView />) //Menuview es el nombre del componente
    const buttonComponent = wrapper.find('button') //indico que busco un boton dentro del componente
    buttonComponent.filter('.btn-aside').simulate('click') //'.btn-aside' es  la clase de mi boton
    expect(wrapper.state); 
});


  describe('CategoryBtn', () => {
    it('Deber√≠a existir el componente <CategoryBtn />', () => {
      const wrapper = shallow(<CategoryBtn/>);
      expect(wrapper.exists()).toBe(true);
    });
});

it('se renderea correctamente el componente OrderTemplate', () => {  
  const component = shallow(<OrderTemplate />);
  expect(toJson(component)).toMatchSnapshot();
});
it('se renderea correctamente el componente OrderRecordTemplate', () => {  
  const component = shallow(<OrderRecordTemplate />);
  expect(toJson(component)).toMatchSnapshot();
});
it('se renderea correctamente el componente OrderRecordTemplate', () => {  
  const component = shallow(<OrderDeliveryTemplate />);
  expect(toJson(component)).toMatchSnapshot();
});
it('se renderea correctamente el componente Navbar', () => {  
  const component = shallow(<Navbar />);
  expect(toJson(component)).toMatchSnapshot();
});
it('se renderea correctamente el componente Btn', () => {  
  const component = shallow(<Btn name="hola"/>);
  expect(toJson(component)).toMatchSnapshot();
});

