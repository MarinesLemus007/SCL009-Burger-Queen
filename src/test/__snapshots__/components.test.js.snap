// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`se renderea correctamente el componente Btn 1`] = `
<button
  className="btn-item"
  onClick={[Function]}
>
  HOLA
</button>
`;

exports[`se renderea correctamente el componente Navbar 1`] = `
<header>
  <nav
    className="menu-nav-row"
  >
    <div
      className="menu-left-col"
    >
      <img
        alt="logo"
        className="menu-icon"
        src="logo.png"
      />
    </div>
    <div
      className="menu-center-col"
    >
      <ul
        className="menu-ul-row"
      >
        <li
          className="menu-ul"
        >
          <Link
            to="/"
          >
            MENÃš
          </Link>
        </li>
        <li
          className="menu-ul"
        >
          <Link
            to="/Kitchen"
          >
            COCINA
          </Link>
        </li>
        <li
          className="menu-ul"
        >
          <Link
            to="/OrdersReady"
          >
            PEDIDOS LISTOS
          </Link>
        </li>
        <li
          className="menu-ul"
        >
          <Link
            to="/Record"
          >
            HISTORIAL
          </Link>
        </li>
      </ul>
    </div>
    <div
      className="menu-right-col"
    />
  </nav>
</header>
`;

exports[`se renderea correctamente el componente OrderRecordTemplate 1`] = `<p />`;

exports[`se renderea correctamente el componente OrderRecordTemplate 2`] = `<p />`;

exports[`se renderea correctamente el componente OrderTemplate 1`] = `<p />`;
